### DISCUSSION TITLE
Command Bots via GUID?

### DISCUSSION THREAD

**Discussion by Russianranger** [trickerer/Trinity-Bots#490] [Ideas]
I don't know how feasible this is, but for macros with commands (ex npcb co unb/reb), would it be possible to enter in a bot GUID vs the name? 

The reason is; current state with my bot addition sql files, there are duplicate names. If you try to utilize commands with two hired bots with the same name, say Arena, it prevents player from unbinding/rebinding the "duplicate". If we can utilize GUID as a means it'll allow more precise control of bots without players having to ensure they have unique names. 

Again - not sure if this is a Herculean effort or if we can implement an OR feature in the code when commanding bots.

**Comment by trickerer** [comment]
No, bots **have** to have unique names to not cause problems to the client. And not only they have to have no collision with one another but with player names too.
The main reason is that client relies of name uniqueness in many aspects, you can see that in group. If you hire Miles the rogue and Miles the priest party will show a duplicated member - whichever was invited first.
There was a fix to make bind commands process bots of the same name properly but that's a temporary solution.
There are no check for name collisions currently, but they will be added eventually.

BTW, you probably mean ID, not GUID. Players aren't supposed to know GUID of anything.

**Reply by Russianranger** [reply]
Got it, and yes you are right, I meant ID. That makes sense, thank you for clarifying. Looks like I'll need to figure out a method to produce some names then. Will try to get ~6000 unique names for bots and refine the SQL addition a bit more. Man that's going to be a pain.

**Comment by Russianranger** [comment]
I figured out a solution - used a python script to append vowels to each bots name, and cycle through 20 times - started with the basic a-e-i-o-u, then added ae, ei, ie, ue, etc. Now each bot has a unique name in the script.
